aspects/aspects.go

```diff
@@ -228,6 +237,91 @@ type accessPattern struct {
 	access accessType
 }
 
+// getPath returns a path if the specified name matches the access pattern's name
+// (with optional placeholders) and an empty string, if not.
+func (p *accessPattern) getPath(name string) (string, error) {
+	if strings.ContainsAny(p.name, "{}") {
+		matches := matchedPlaceholders(p.name, name)
+		if matches == nil {
+			// aspect name w/ placeholders doesn't match the specified name
+			return "", nil
+		}
+
+		return fillInPlaceholders(p.path, matches)
+	} else if p.name == name {
+		return p.path, nil
+	}
+
+	return "", nil
+}
+
+// matchedPlaceholders takes a dot-separated pattern with optional placeholders
+// and a name. To match the pattern, the name's dot-separated parts must be equal
+// to the pattern's non-placeholder parts. It returns a map from placeholders to
+// their matches in the name (can be empty, if the path has no placeholders). If
+// there's no match, it returns nil. For example, if pattern="{foo}.b.{bar}" and
+// name="a.b.c", then it returns {"foo": "a", "bar": "c"}.
+func matchedPlaceholders(pattern, name string) map[string]string {
+	patternParts, nameParts := strings.Split(pattern, "."), strings.Split(name, ".")
+
+	if len(patternParts) != len(nameParts) {
+		return nil
+	}
+
+	placeholderMatches := make(map[string]string)
+	for i, patt := range patternParts {
+		// TODO: check assertions for:
+		//	* empty name parts (e.g., "a..b")
+		//	* names with non-ASCII characters (or we can't index the name directly)
+		//	* mismatched or out-of-place curly brackets (assuming they can only be used for placeholders)
+		//	* placeholders in a key but not in the path and vice-versa
```

Comment:
most of these checks should be done when we build the directory, we probably want to precompute the splittings and a few things when we build the accessPattern, have a newAccessPattern and for example have a field for the exploded name pattern like:

```
nameMatchers []nameMatcher
```
and nameMatcher is an interface implemented on top of string corresponding to the pattern parts:

```
type nameMatcher struct{
   match(namePart string, placeholders map[string]string) bool
 }
```

or similar.

Something similar can be done with the path with a `pathFiller` interface using the `strings.Builder`.

Reply:
I'm done with the validation part but it adds a fair amount of code even without the tests so it might be better to do as a follow-up. I think the precomputation will be similar so it's probably better to have three ~225 line PRs that are cleanly split rather than one 700 line PR.